extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/linkHandler/YoutubeStreamLinkHandlerFactory.java

```diff
@@ -46,13 +48,25 @@ public static YoutubeStreamLinkHandlerFactory getInstance() {
         return instance;
     }
 
+    private static String extractId(@Nullable String id) {
+        if (id != null) {
+            Matcher m = Pattern.compile("([a-zA-Z0-9_-]{11})").matcher(id);
+            return m.find() ? m.group(1) : "";
+        }
+        return "";
+    }
+
     private static boolean isId(@Nullable String id) {
```

Comment:
We might consider removing `isId()` altogether and replacing
```java
if (isId()) {
    return extractId()
}
```
with 
```java
final String extractedId = extractId(id);
if (extractedId != null) {
    return extractedId;
}
```
which does one regex comparison less, and is even clearer imo